File: Object Oriented Programming/learning/classes_and_objects/exercises/cpp/ex-13.cpp
/*
Write a program by creating an 'Employee' class 
having the following functions and print the final salary.
1 - 'getInfo()' which takes the salary, number of hours of work per day of employee as parameters
2 - 'AddSal()' which adds $10 to the salary of the employee if it is less than $500.
3 - 'AddWork()' which adds $5 to the salary of the employee if the number of hours of work per day is more than 6 hours.
*/
#include <iostream>
#include <string>
using namespace std;

// define class
class Employee {
    private:
        double salary, num_hours;

    public:
        Employee(double sal, double num):
        salary(sal), num_hours(num) {}

        void getInfo(double sal, double num) {
            salary = sal;
            num_hours = num;
        }

        void AddSal() {
            if (salary < 500) {
                salary += 10;
            }
        }

        void AddWork() {
            if (num_hours > 6) {
                salary += 5;
            }
        }

        double getSalary() {
            return salary;
        }
};

int main() {
    // create object of class
    Employee emp(0, 0);
    
    // initialize variables
    double initialSalary, hoursWork;

    // take input from user
    cout<<"Enter Salary: ";
    cin>>initialSalary;
    cout<<"Enter Number of working hours: ";
    cin>>hoursWork;

    // input initial info
    emp.getInfo(initialSalary, hoursWork);

    // add salary and then add add work
    emp.AddSal();
    emp.AddWork();

    // display finaly salary
    cout<<"Final salary: $"<<emp.getSalary()<<endl;

    return 0;
}

File: Object Oriented Programming/learning/classes_and_objects/exercises/py/ex-13.py
'''
Write a program by creating an 'Employee' class 
having the following functions and print the final salary.
1 - 'getInfo()' which takes the salary, number of hours of work per day of employee as parameters
2 - 'AddSal()' which adds $10 to the salary of the employee if it is less than $500.
3 - 'AddWork()' which adds $5 to the salary of the employee if the number of hours of work per day is more than 6 hours.
'''

# define a class
class Employee:
    # initialize constructor
    def __init__(self, salary, num_hours):
        self.__salary = salary
        self.__num_hours = num_hours

    def getInfo(self, salary, num_hours):
        self.__salary = salary
        self.__num_hours = num_hours

    def AddSal(self):
        if self.__salary < 500:
            self.__salary += 10

    def AddWork(self):
        if self.__num_hours > 6:
            self.__salary += 5

    def getSalary(self):
        return self.__salary

# create object of class
emp = Employee(0, 0)

# take input from user
initialSalary = float(input("Enter Salary: "))
hoursWorked = int(input("Enter number of hours of work per day: "))

# call member functions
emp.getInfo(initialSalary, hoursWorked)
emp.AddSal()
emp.AddWork()

print(f"Final Salary: {emp.getSalary()}")
